core/src/test/java/org/infinispan/reactive/publisher/impl/SimpleLocalPublisherMangerTest.java

```diff
@@ -94,4 +104,65 @@ public void testNoIntermediateOps(DeliveryGuarantee deliveryGuarantee, boolean i
 
       results.forEach(assertConsumer);
    }
+
+   @DataProvider(name = "GuaranteeParallelEntry")
+   public Object[][] deliveryGuaranteeParallelEntryProvider() {
+      return Arrays.stream(DeliveryGuarantee.values())
+            .flatMap(dg -> Stream.of(Boolean.TRUE, Boolean.FALSE)
+                  .flatMap(parallel -> Stream.of(Boolean.TRUE, Boolean.FALSE)
+                        .map(entry -> new Object[]{dg, parallel, entry}))
+            )
+            .toArray(Object[][]::new);
+   }
+
+   @Test(dataProvider = "GuaranteeParallelEntry")
+   public void testWithAsyncOperation(DeliveryGuarantee deliveryGuarantee, boolean isParallel, boolean isEntry) {
+      Cache<Integer, String> cache = cache(0);
+      Map<Integer, String> inserted = insert(cache);
+
+      BlockingManager blockingManager = TestingUtil.extractComponent(cache, BlockingManager.class);
+
+      LocalPublisherManager<Integer, String> lpm = lpm(cache);
+      IntSet allSegments = IntSets.immutableRangeSet(SEGMENT_COUNT);
+      CompletionStage<PublisherResult<Set<Object>>> stage;
+      Consumer<Object> assertConsumer;
+      Collector<Object, ?, Set<Object>> collector = Collectors.toSet();
+      BiFunction<Set<Object>, Set<Object>, Set<Object>> reduceBiFunction = (left, right) -> {
+         left.addAll(right);
+         return left;
+      };
+
+      io.reactivex.rxjava3.functions.Function<Object, Single<Object>> sleepOnBlockingPoolFunction = value ->
+            Single.fromCompletionStage(blockingManager.supplyBlocking(() -> value, "test-blocking-thread"));
+
+      if (isEntry) {
+         stage = lpm.keyReduction(isParallel, allSegments, null, null, false,
+               deliveryGuarantee, publisher -> Flowable.fromPublisher(publisher)
+                     .concatMapSingle(sleepOnBlockingPoolFunction).collect(collector).toCompletionStage()
+               , publisher -> Flowable.fromPublisher(publisher)
+                     .reduce(reduceBiFunction).toCompletionStage(Collections.emptySet()));
```

Comment:
Nitpicking maybe, but the orphan `,` really bugs me. I like the one line per operation format much better, although in the suggestion box it looks worse.

```suggestion
         stage = lpm.keyReduction(isParallel, allSegments, null, null, false, deliveryGuarantee,
                                  publisher -> Flowable.fromPublisher(publisher)
                                                       .concatMapSingle(sleepOnBlockingPoolFunction)
                                                       .collect(collector)
                                                       .toCompletionStage(),
                                  publisher -> Flowable.fromPublisher(publisher)
                                                       .reduce(reduceBiFunction)
                                                       .toCompletionStage(Collections.emptySet()));
```

Reply:
Sure, I will see how it comes out.